apps/passport-server/src/services/telegramService.ts

```diff
@@ -227,11 +227,30 @@ export class TelegramService {
                 : null
             };
           });
-          const eventsWithChats = await Promise.all(eventsWithChatsRequests);
+          const eventsWithChatsSettled = await Promise.allSettled(
+            eventsWithChatsRequests
+          );
+          const eventsWithChats = eventsWithChatsSettled
+            .filter((e) => e.status == "fulfilled")
+            // @ts-expect-error value after filtering for success
+            .map((e) => e.value);
+
+          if (eventsWithChats.length === 0) {
+            return ctx.api.editMessageText(
+              userId,
+              msg.message_id,
+              `No chats found to link with. Try again soon!`,
+              {
+                parse_mode: "HTML"
+              }
+            );
+          }
+
           let eventsHtml = `<b> Current Chats with Events </b>\n\n`;
+
           for (const event of eventsWithChats) {
-            // @ts-expect-error chat title
-            eventsHtml += `Event: <b>${event.eventName}</b> ➡ Chat: <i>${event.chat?.title}</i>\n`;
+            if (event.chat?.title)
```

Comment:
curious - when would this be true? if `title` isn't defined but the chat is, i'd rather it use some other identifier than skip it entirely

Reply:
Chat is undefined if the bot is not an admin of those chats. Title is undefined if the chat doesn't have a title (ie, a it's a DM). We need both to be true to show if to the user